### PR TITLE
[Commands]: Fix NotReached() crash

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/37275

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue

### CODE REVIEW

File: patches/ui-base-accelerators-accelerator.cc.patch [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 +++ b/ui/base/accelerators/accelerator.cc
-@@ -388,6 +388,7 @@ std::u16string Accelerator::ApplyLongFormModifiers(
-   // Note that we use 'else-if' in order to avoid using Ctrl+Alt as a shortcut.
-   // See https://devblogs.microsoft.com/oldnewthing/20040329-00/?p=40003 for
-   // more information.
+@@ -380,6 +380,7 @@ std::u16string Accelerator::KeyCodeToName() const {
+ 
+ std::u16string Accelerator::ApplyLongFormModifiers(
+     const std::u16string& shortcut) const {
 +  BRAVE_UI_BASE_ACCELERATOR_APPLY_LONG_FORM_MODIFIERS_
```
Comment (fallaciousreasoning) [inline patches/ui-base-accelerators-accelerator.cc.patch:9]: Moved this to the start of the function, now we don't reuse any of the chromium code.

**Review by simonhong — APPROVED** [review]
++ with one nits 👍🏼

**Review by mkarolin — APPROVED** [review]
`chromium_src` and `strings` ++

File: chromium_src/ui/base/accelerators/accelerator.cc [hunk 2]
```diff
@@ -5,11 +5,57 @@
 
 #include "ui/base/accelerators/accelerator.h"
 
+#include <string>
+
+#include "components/grit/brave_components_strings.h"
```
Comment (mkarolin) [inline chromium_src/ui/base/accelerators/accelerator.cc:10]: Do we need to add a dependency for this to ui/base/BUILD.gn?

Comment (simonhong) [reply chromium_src/ui/base/accelerators/accelerator.cc:10]: > Okay, this took me quite a while to work out - gn_check was passing, which was a good sign:

I think I missed this. AFAIK, `gn_check` doesn't check deps for files under`chromium_src`.
So, we have to be careful whenever including new headers in `chroium_src` cc @goodov 
Our rule is adding all deps in BUILD.gn directly. so I think maybe we should have patching in `ui/base/BUILD.gn` something like this?
```
deps += brave_chromium_src_ui_base_deps
```

File: patches/ui-base-BUILD.gn.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/base/BUILD.gn b/ui/base/BUILD.gn
+index 3452178f0db87..7b9ede7f64974 100644
+--- a/ui/base/BUILD.gn
++++ b/ui/base/BUILD.gn
+@@ -552,6 +552,7 @@ component("base") {
+       "//ui/events",
+       "//ui/events/devices",
+     ]
++    import("//brave/chromium_src/ui/base/sources.gni") deps += brave_chromium_src_ui_base_deps
```
Comment (bridiver) [inline patches/ui-base-BUILD.gn.patch:9]: this is inside `use_blink` which is wrong and it fails on ios. Also the correct dep to use is `//brave/components/resources:strings`

**Pull request merged**